src/pages/ReimbursementAccount/BeneficialOwnerInfo/BeneficialOwnerInfo.tsx

```diff
@@ -27,102 +22,82 @@ import LegalNameUBO from './substeps/BeneficialOwnerDetailsFormSubsteps/LegalNam
+        beneficialOwnerKeys: reimbursementAccount?.achData?.beneficialOwnerKeys ?? reimbursementAccountDraft?.beneficialOwnerKeys ?? [],
     };
 
     // We're only reading beneficialOwnerKeys from draft values because there is not option to remove UBO
     // if we were to set them based on values saved in BE then there would be no option to enter different UBOs
     // user would always see the same UBOs that was saved in BE when returning to this step and trying to change something
-    const [beneficialOwnerKeys, setBeneficialOwnerKeys] = useState(defaultValues[BENEFICIAL_OWNER_KEYS_KEY]);
+    const [beneficialOwnerKeys, setBeneficialOwnerKeys] = useState<string[]>(defaultValues.beneficialOwnerKeys);
     const [beneficialOwnerBeingModifiedID, setBeneficialOwnerBeingModifiedID] = useState('');
     const [isEditingCreatedBeneficialOwner, setIsEditingCreatedBeneficialOwner] = useState(false);
-    const [isUserUBO, setIsUserUBO] = useState(defaultValues[BENEFICIAL_OWNER_INFO_STEP_KEYS.OWNS_MORE_THAN_25_PERCENT]);
-    const [isAnyoneElseUBO, setIsAnyoneElseUBO] = useState(defaultValues[BENEFICIAL_OWNER_INFO_STEP_KEYS.HAS_OTHER_BENEFICIAL_OWNERS]);
+    const [isUserUBO, setIsUserUBO] = useState(defaultValues.ownsMoreThan25Percent);
+    const [isAnyoneElseUBO, setIsAnyoneElseUBO] = useState(defaultValues.hasOtherBeneficialOwners);
     const [currentUBOSubstep, setCurrentUBOSubstep] = useState(1);
     const canAddMoreUBOS = beneficialOwnerKeys.length < (isUserUBO ? MAX_NUMBER_OF_UBOS - 1 : MAX_NUMBER_OF_UBOS);
 
     const submit = () => {
-        const bankAccountID = getDefaultValueForReimbursementAccountField(reimbursementAccount, 'bankAccountID', 0);
-
-        const beneficialOwners =
-            beneficialOwnerKeys.length === 0
-                ? []
-                : _.map(beneficialOwnerKeys, (ownerKey) => ({
-                      firstName: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_firstName`),
-                      lastName: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_lastName`),
-                      dob: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_dob`),
-                      ssnLast4: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_ssnLast4`),
-                      street: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_street`),
-                      city: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_city`),
-                      state: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_state`),
-                      zipCode: lodashGet(reimbursementAccountDraft, `beneficialOwner_${ownerKey}_zipCode`),
-                  }));
+        const beneficialOwnerFields = ['firstName', 'lastName', 'dob', 'ssnLast4', 'street', 'city', 'state', 'zipCode'];
```

Comment:
minor but maybe you can move it outside of component scope?

Reply:
It's strict connected with this function, so I'm not sure what benefits we have move it out of the component. This is simple array, so shouldn't consume a lot of memory ;)